src/core/Akka/Actor/ActorRefProvider.cs

```diff
@@ -155,7 +156,14 @@ public LocalActorRefProvider(string systemName, Settings settings, EventStream e
             _tempNumber = new AtomicCounterLong(1);
             _tempNode = _rootPath / "temp";
 
-            //TODO: _guardianSupervisorStrategyConfigurator = dynamicAccess.createInstanceFor[SupervisorStrategyConfigurator](settings.SupervisorStrategyClass, EmptyImmutableSeq).get
+            var supervisorStrategyType = Type.GetType(settings.SupervisorStrategyClass);
+            if (supervisorStrategyType == null)
+            {
+                // TODO: is it ok to throw exception here?
+                throw new ArgumentException(string.Format("Type \"{0}\" is not found.", settings.SupervisorStrategyClass));
```

Comment:
We throw `Akka.ConfigurationException` if the source of the value is the HOCON config.
`ArgumentException` is more of a user provided value.

There is also a third option here and use `Type.GetType(settings.SupervisorStrategyClass,true)`
Which will throw one of the following exceptions : https://msdn.microsoft.com/en-us/library/a87che9d(v=vs.110).aspx 

I am leaning towards the latter as it would be the most idiomatic .NET way to deal with missing types